Integration

frankliu81 opened the pull request.

Commit:
added initial blog models

Commit:
added crud for blog posts

Commit:
added crud for comments

Commit:
added rspec test for posts

Commit:
updated my incorrect migrations for including post_id in comment and category, had to drop database, ok for personal test project

Commit:
added rspec unit test for post model and controller and factory_girl

Commit:
added product references to comments and the associations, need to update post show

frankliu81 committed:
generated 101 posts [seeds_update]

Commit:
generated 102 posts [seeds_update_2]

Commit:
added application.css.scss

Commit:
merge seeds.rb conflict

frankliu81 committed:
seed blog tool with posts and comments

frankliu81 committed:
added partial view _comment.html.erb, and remove existing comment actions and reimplement create and destroy per the new embedded comment routes

frankliu81 committed:
updated gem to include pry-byebug

frankliu81 committed:
added category, changed the create category migration to use name field, show category and allow for selection on post create

frankliu81 committed:
added user authentication to blog

frankliu81 committed:
added user profile and password update, and user delete with session clear

frankliu81 committed:
added ability to change user password with old password, new password, new password confirmation form; added separate form for updating the user profile only

frankliu81 committed:
added user authorization, only admin and user who created the post/comment can edit or delete

frankliu81 committed:
add mailer, delayed_job, and password reset

frankliu81 committed:
add comments mailer and send a mail to the post owner every time a comment is added

frankliu81 committed:
refactor users/password_changes to its own controller

frankliu81 committed:
add email account verification during sign up

frankliu81 committed:
update seeds.rb with associations for comment's and comment's posts; update factories for user and posts with associations; update existing rspec test to make them pass with sign_in, need to add not signed in test

frankliu81 committed:
add rspec tests for posts user authentication and user authorization

frankliu81 committed:
add users posts favorites many-to-many

frankliu81 committed:
add posts tags taggings many-to-many

frankliu81 committed:
fix indentation error with posts_controller_spec #destroy

frankliu81 committed:
add navbar

frankliu81 committed:
add search, pagination, and new form styling with bootstrap

frankliu81 committed:
style forms with bootstrap, add hover slideDown with jquery for navbar

frankliu81 committed:
fix bug with pasing in empty password for reset, refactor password change into user model's attempt_change_password

frankliu81 committed:
update account verification to use account verification token, also make edit action activate the user account without an extra click for update

frankliu81 committed:
update rake task name to use the blog namesapce

frankliu81 committed:
add google oauth and allow for google sign-in from sign-in page

frankliu81 committed:
integrate jquery tag-it to the post tagging feature

frankliu81 committed:
make the post show page display tag-it tags as well

frankliu81 committed:
AJAXify comment create and destroy

frankliu81 committed:
AJAXify comment edit and update

frankliu81 committed:
AJAXIfy post favorite

frankliu81 committed:
fix bug with favorited posts, should not do find on post_id during index

frankliu81 committed:
add active serializer and blog API for index and show

frankliu81 committed:
add base api controller to set session to null on forgery protection, api to create post

frankliu81 committed:
Fix typo bug in comment ajax

lucasckw2022 commented on config/routes.rb:
```diff
@@ -1,10 +1,65 @@
+  #get "/auth/:provider/callback" => 'sessions#create'
+  get "/auth/google_oauth2/callback" => 'callbacks#google'
 
+  root "home#index", as: :root
```
maybe as: :root is not necessary since you have used root_path in controllers

lucasckw2022 commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,59 @@
+  def edit
+    #puts params
+
+    @user = User.find_by_id params[:id]
```
Since .find is already finding its id, maybe could consider using User.find params[:id] instead? But it works either way

lucasckw2022 commented on app/controllers/posts_controller.rb:
```diff
@@ -0,0 +1,98 @@
+      redirect_to post_path(@post), notice: "Post created"
+    else
+      #render text: "FAILURE"
+      flash[:alert] = "Post not created"
```
Since you have the counting of error message and errors. Maybe it is not necessary to have this alert? Maybe consider adding this to to your counting instead, like: "Post not cretaed. The form contains 3 errors" ?

lucasckw2022 commented on app/assets/stylesheets/posts.scss:
```diff
@@ -0,0 +1,26 @@
+}
+
+.post-comment__textarea {
+  height: 80% !important;
```
suggested that setting height in actual px instead of %, since simple form seems to generate a div to wrap the fields with errors after validation. Since the height of its parent has changed, the height of the input box will change as well

lucasckw2022 commented on app/views/comments/_form.html.erb:
```diff
@@ -0,0 +1,13 @@
+<%= @comment.errors.full_messages.join(", ") %>
+
+<%= form_for @comment, url: post_comments_path(@post), html: {class: "form"}, remote: true do |f| %>
```
Need to change the @comment to comment since it is being assigned to be "comment: Comment.new" in create_success.js.erb. Otherwise, the input field would not be rendered.

frankliu81 committed:
incorporate code review feedback from Lucas
update .post-comment__textarea css to use fixed height so it doesn't resize after simple form errors
fix _form.html.erb partial to use comment instead of @comment because we passed in the comment variable to the partial

frankliu81 committed:
Remove unnecessary as: :root in routes.rb
